OpenRA.Mods.Common/Widgets/Logic/Editor/MapEditorLogic.cs

```diff
@@ -47,9 +59,9 @@ public MapEditorLogic(Widget widget, World world, WorldRenderer worldRenderer)
 					return item;
 				};
 
-				var options = new[] { 2f, 1f, 0.5f, 0.25f };
+				var options = worldRenderer.Viewport.AvailableZoomSteps;
 				zoomDropdown.OnMouseDown = _ => zoomDropdown.ShowDropDown("LABEL_DROPDOWN_TEMPLATE", 150, options, setupItem);
-				zoomDropdown.GetText = () => selectedLabel;
+				zoomDropdown.GetText = () => zoomDropdown.SelectedItem.ToString();
```

Comment:
Keep this as `selectedLabel` please.  This gets invoked every render frame, so creates a lot of unnecessary objects.